odoo/addons/base/models/ir_qweb.py

```diff
@@ -224,6 +286,55 @@ def _compile_directive_call_assets(self, el, options, indent):
 
     # method called by computing code
 
+    def _start_profiling(self, directive, xpath, options):
+        data = [None, None]
+
+        if self.env.context.get('profile_execution_context_qweb'):
+            # Add some context on thread at current call stack level.
+            # This context stored by collector beside stack and is used by
+            # Speedscope to add a level to the stack with this QWeb directive
+            # information.
+            current_thread = threading.current_thread()
+            previous_context = getattr(current_thread, 'exec_context', ())
+            context = {'directive': directive, 'xpath': xpath}
+            current_thread.exec_context = previous_context + ((tools.profiler.stack_size(), context),)
+            data[0] = previous_context
```

Comment:
I would prefer to keep execution context structure logic into the profiler files.

Why not something like
```
execution_context = tools.profiler.ExecutionContext()
execution_context.__enter__()
```
But you may have the offset problem again, to check